typings/styled-components.d.ts

```diff
@@ -20,7 +20,7 @@ export type OuterStyledProps<P> = ThemedOuterStyledProps<P, any>;
 
 export type Interpolation<P> = FlattenInterpolation<P> | ReadonlyArray<FlattenInterpolation<P> | ReadonlyArray<FlattenInterpolation<P>>>;
 export type FlattenInterpolation<P> = InterpolationValue | InterpolationFunction<P>;
-export type InterpolationValue = string | number;
+export type InterpolationValue = string | number | StyledComponentClass<any, any>;
```

Comment:
It can. So `StyledComponentClass` produced by a styled function:
```ts
const Link: StyledComponentClass<any, any> = styled.a` color: blue; `;
```
then it can be used in both functional and constant interpolations:
```ts
const Section = styled.section`
  color: black;
  ${Link} {
    color: red;
  }
  ${p => p.theme.linkClass || Link} {
    color: red;
  }
`;
```